build.gradle

```diff
@@ -48,33 +56,44 @@ def attrs = [
 
 tasks.withType(AsciidoctorTask) { task ->
     attributes attrs
+    sourceDir 'src/docs/asciidoc'
+    baseDirFollowsSourceDir()
     sources {
         include 'index.adoc'
     }
 }
 
 task html(type: AsciidoctorTask, description: 'Generates single page HTML') {
-    backends 'html5'
+    outputDir "$buildDir/asciidoc/html5"
+    outputOptions {
+        backends 'html5'
+    }
 }
 
 task prepress(type: AsciidoctorTask, description: 'Generates PDF for prepress printing') {
     attributes attrs + ['media': 'prepress', 'pdfmarks': '', 'pdf-theme': 'prepress', 'source-highlighter': '']
-    requires file('src/main/ruby/asciidoctor-pdf-extensions.rb')
-    backends 'pdf'
     outputDir "$buildDir/asciidoc/pdf-prepress"
-    separateOutputDirs false
+    outputOptions {
+        backends 'pdf'
+        setSeparateOutputDirs false
+    }
 }
 
 task pdf(type: AsciidoctorTask, description: 'Generates PDF for download') {
     attributes attrs + ['pdfmarks': '']
-    requires file('src/main/ruby/asciidoctor-pdf-extensions.rb')
     outputDir "$buildDir/asciidoc/pdf-screen"
-    backends 'pdf'
-    separateOutputDirs false
+    outputOptions {
+        backends 'pdf'
+        setSeparateOutputDirs false
+    }
 }
 
 task epub(type: AsciidoctorTask, description: 'Generates EPUB3') {
-    backends 'epub3'
+    attributes attrs + ['source-highlighter': 'coderay']
```

Comment:
@Mogztter I found this is necessary to get *any* syntax highlighting in EPUB. If I use `rouge`, there's no highlighting. With `coderay`, only HTML is highlighted. Ideally, it'd be possible to get TypeScript and Kotlin highlighting in EPUB. `rouge` works excellent for PDF and HTML.